sw/host/opentitanlib/src/tpm/driver.rs

```diff
@@ -233,20 +265,35 @@ impl SpiDriver {
         self.spi
             .run_transaction(&mut [spi::Transfer::Both(&req, &mut buffer)])?;
         if buffer[3] & 1 == 0 {
-            let mut retries = 10;
+            let start_time = SystemTime::now();
             while {
                 self.spi
                     .run_transaction(&mut [spi::Transfer::Read(&mut buffer[0..1])])?;
                 buffer[0] & 1 == 0
             } {
-                retries -= 1;
-                if retries == 0 {
+                if SystemTime::now().duration_since(start_time)?.cmp(&TIMEOUT) == Ordering::Greater
```

Comment:
`Duration` implements `PartialOrd`, which means you should be able to use the arithmetic operators here:
```
if SystemTime::now().duration_since(start_time)? > TIMEOUT { ... }
```

Reply:
Done